.github/workflows/build-test.yml

```diff
@@ -563,14 +563,108 @@ jobs:
+      with:
+        path: |
+          ~/.cargo/bin/cargo-make
+          ~/.cargo/bin/cargo-make.exe
+        key: ${{ runner.os }}-${{ steps.cargo-make-version.outputs.hash }}
+
+    - name: Install cargo-make
+      if: steps.cargo-make-cache.outputs.cache-hit != 'true'
+      uses: actions-rs/install@v0.1.2
+      with:
+        crate: cargo-make
+        version: latest
+
+    - name: Install prerequisites for wasm build
+      run: |
+        rustup component add rust-src
+        rustup toolchain list
+        rustup toolchain install nightly-2021-02-28
+        sudo npm install -g wasm-opt --unsafe-perm
+        sudo npm install -g wabt
+        cargo install twiggy
+
+    - name: Setup output data directory
+      run: mkdir -p benchmarks/binsize
+
+    - name: Build wasm executables and measure size
+      run: |
+        cargo make wasm-examples
+        cargo run --package icu_benchmark_binsize | tee benchmarks/binsize/output.txt
+
+    - name: Store benchmark result (non-merge action only)
+      # Data from anything that is not a merge to mainline goes to a preliminary branch
+      if: github.event_name != 'push' || github.ref != 'refs/heads/main'
+      # Use gregtatum special feature to process ndjson-formatted benchmark data
+      uses: gregtatum/github-action-benchmark@d3f06f738e9612988d575db23fae5ca0008d3d12
+      with:
+        tool: 'ndjson'
+        output-file-path: benchmarks/binsize/output.txt
+        benchmark-data-dir-path: ./benchmarks/binsize
+        # Tentative setting, optimized value to be determined
+        alert-threshold: '200%'
+        fail-on-alert: true
+        gh-pages-branch:  unmerged-pr-bench-data
+        auto-push: false
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        comment-on-alert: true
+
+    - name: Store benchmark result (merge to main only)
+      # Only for PRs that merge into the ICU4X mainline.
+      if: github.event_name == 'push' && github.ref == 'refs/heads/main' && github.repository == 'unicode-org/icu4x'
+      # Use gregtatum special feature to process ndjson-formatted benchmark data
+      uses: gregtatum/github-action-benchmark@d3f06f738e9612988d575db23fae5ca0008d3d12
+      with:
+        tool: 'ndjson'
+        output-file-path: benchmarks/binsize/output.txt
+        benchmark-data-dir-path: ./benchmarks/binsize
+        # Tentative setting, optimized value to be determined
+        alert-threshold: '200%'
+        fail-on-alert: true
+        gh-pages-branch: gh-pages
+        auto-push: true
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        comment-on-alert: true
+        alert-comment-cc-users: '@gnrunge,@sffc,@zbraniecki,@echeran,@gregtatum'
+
+    - name: Switch branch
+      # Only for PRs that merge into the ICU4X mainline
+      if: github.event_name == 'push' && github.ref == 'refs/heads/main' && github.repository == 'unicode-org/icu4x'
+      run: git checkout unmerged-prpbench-data # gh-pages
+
+    - name: Upload binsize with ndjson data
```

Comment:
suggestion(optional): reword job name for clarity

maybe "Upload binsize benchmark data in njdson format (merge to main only)"

The first part is to clarify that the data is the binsize. njdson is just the format of the data. The latter part in parentheses is for the same reason as above.

Reply:
Done.